.pipelines/UpdateVersions.ps1

```diff
@@ -0,0 +1,126 @@
+Param(
+# Using the default value of 1.6 for versionNumber and useExperimentalVersion as false
+  #[Parameter(Mandatory=$False,Position=1)]
+  [string]$versionNumber = "1.6",
+
+  #[Parameter(Mandatory=$False,Position=2)]
+  [string]$useExperimentalVersion = "false"
```

Comment:
Any reason to be string instead of boolean?

Reply:
When the pipeline calls the PS1 file, the passed parameters are converted to string type.

Reply:
That is unexpected.

You can pass any type of PowerShell object to a script in a pipeline.

```yaml
pwsh: |-
  & ./Path/To/Script.ps1 -BooleanArgunent $true
```

If you need to convert an azure DevOps yaml boolean into a PowerShell boolean, you can use...

```yaml
$${{parameters.azureDevopsYamlBoolean}}
```

This will evaluate to `$true` or `$false` (the first $ is for PowerShell, and the second one is for Azure DevOps!)